### PR TITLE
Add json config

### PR DESCRIPTION
**Pull request by avidraccoon**
Adds ability to load constants from JSON file.

### CODE REVIEW

File: parameter_tools/build.gradle
Comment (jkleiber) [inline parameter_tools/build.gradle:1]: Can we add the `spotless` formatter to this build.gradle so the code gets auto-formatted?

File: parameter_tools/src/main/java/coppercore/parameter_tools/JSONSync.java [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+        this.file = file;
+    }
+
+    private static record JSONSyncConfig(boolean serializeNulls, boolean prettyPrinting, boolean excludeFieldsWithoutExposeAnnotation, FieldNamingPolicy namingPolicy,
+    ToNumberPolicy numberToNumberPolicy, ToNumberPolicy objectToNumberPolicy, LongSerializationPolicy longSerializationPolicy, boolean autoReload) {       
+        public JSONSyncConfig(JSONSyncConfigBuilder builder){
+            this(builder.serializeNulls, builder.prettyPrinting, builder.excludeFieldsWithoutExposeAnnotation, builder.namingPolicy, builder.numberToNumberPolicy,
+                builder.objectToNumberPolicy, builder.longSerializationPolicy, builder.autoReload);
+        }
+    }
```
Comment (jkleiber) [inline parameter_tools/src/main/java/coppercore/parameter_tools/JSONSync.java:55]: Why is there a method within a method here?

**Review by jkleiber — COMMENTED** [review]
First pass review

File: parameter_tools/src/main/java/coppercore/parameter_tools/JSONSync.java (hunk 1 above)
Comment (avidraccoon) [reply parameter_tools/src/main/java/coppercore/parameter_tools/JSONSync.java:55]: One is a record which is a data structure that is immutable and the other is a constructor for the record.

File: parameter_tools/build.gradle
Comment (avidraccoon) [reply parameter_tools/build.gradle:1]: I believe we can, but I am not sure.

**Pull request merged**